django/db/models/fields/__init__.py

```diff
@@ -242,31 +242,55 @@ def _check_field_name(self):
             return []
 
     def _check_choices(self):
-        if self.choices:
-            if isinstance(self.choices, str) or not is_iterable(self.choices):
-                return [
-                    checks.Error(
-                        "'choices' must be an iterable (e.g., a list or tuple).",
-                        obj=self,
-                        id='fields.E004',
-                    )
-                ]
-            elif any(isinstance(choice, str) or
-                     not is_iterable(choice) or len(choice) != 2
-                     for choice in self.choices):
-                return [
-                    checks.Error(
-                        "'choices' must be an iterable containing "
-                        "(actual value, human readable name) tuples.",
-                        obj=self,
-                        id='fields.E005',
-                    )
-                ]
-            else:
-                return []
+        if not self.choices:
+            return []
+
+        def _is_terminal(value):
```

Comment:
The code might be more readable with a name something like `is_valid_choice` -- does that capture the intent?

Reply:
Not exactly, because we are dealing an element of the choice (either the "actual value to be set on the model" or the "human-readable name" described in https://docs.djangoproject.com/en/dev/ref/models/fields/#django.db.models.Field.choices).

For lack of a better name, I'll go with `is_valid_value`.

Reply:
The name `is_valid_value` isn't great. The first block would read:
```python
        if is_valid_value(self.choices):
            return [checks.Error(...)]
```

I chose `is_value` instead.